Bump version to 0.0.32 (#3302)
astral-sh/ty: CHANGELOG.md

# Changelog

## 0.0.32

Released on 2026-04-20.

### Bug fixes

- Fix panic when `__get__` uses `Concatenate` `self`-type and wraps a `__call__` ([#24692](https://github.com/astral-sh/ruff/pull/24692))
- Avoid panicking on overloaded `Callable` type context ([#24661](https://github.com/astral-sh/ruff/pull/24661))
- Expand class bases in per-base lint checks ([#24695](https://github.com/astral-sh/ruff/pull/24695), [#24699](https://github.com/astral-sh/ruff/pull/24699))
- Fix stack overflow for binary operator inference involving recursive types ([#24551](https://github.com/astral-sh/ruff/pull/24551))

### LSP server

> - Report unreachable code as unnecessary hint diagnostics ([#24580](https://github.com/astral-sh/ruff/pull/24580))

Reviewer: ```suggestion
- Dim out unreachable code in IDEs ([#24580](https://github.com/astral-sh/ruff/pull/24580))
```

- Do not suggest argument completion when typing the value of a keyword argument ([#24669](https://github.com/astral-sh/ruff/pull/24669))

> - Group overloaded methods for docstrings ([#23920](https://github.com/astral-sh/ruff/pull/23920))

Reviewer: ```suggestion
- Retrieve the docstring from the overload implementation if an `@overload`-decorated function has no docstring ([#23920](https://github.com/astral-sh/ruff/pull/23920))
```

### Core type checking

> - Show error context for assignability diagnostics ([#24309](https://github.com/astral-sh/ruff/pull/24309))

Reviewer: should we add a `## Diagnostics` section and put both this and all the changes currently in the `## Other` section together in that section?

- Allow `if` statements in TypedDict bodies ([#24702](https://github.com/astral-sh/ruff/pull/24702))
- Disallow `@disjoint_base` on TypedDicts and Protocols ([#24671](https://github.com/astral-sh/ruff/pull/24671))
- Do not consider a subclass of a `@dataclass_transform`-decorated class to have dataclass-like semantics if it has `type` in its MRO ([#24679](https://github.com/astral-sh/ruff/pull/24679))

> - Use partially qualified names when reporting diagnostics regarding bad calls to methods ([#24560](https://github.com/astral-sh/ruff/pull/24560))

Reviewer: this could also go in a `## Diagnostics` section

# Changelog

## 0.0.32

Released on 2026-04-20.

### Bug fixes

- Fix panic when `__get__` uses `Concatenate` `self`-type and wraps a `__call__` ([#24692](https://github.com/astral-sh/ruff/pull/24692))
- Avoid panicking on overloaded `Callable` type context ([#24661](https://github.com/astral-sh/ruff/pull/24661))
- Expand class bases in per-base lint checks ([#24699](https://github.com/astral-sh/ruff/pull/24699))

> - Expand class bases when reporting diagnostics ([#24695](https://github.com/astral-sh/ruff/pull/24695))

Reviewer 2: ```suggestion
- Expand class bases in per-base lint checks ([#24695](https://github.com/astral-sh/ruff/pull/24695), [#24699](https://github.com/astral-sh/ruff/pull/24699))
```
